src/lib/fields/CustomField.js

```diff
@@ -0,0 +1,69 @@
+// This file is part of React-Invenio-Deposit
+// Copyright (C) 2020 CERN.
+// Copyright (C) 2020 Northwestern University.
+//
+// React-Invenio-Deposit is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+
+import _get from "lodash/get";
+import _set from "lodash/set";
+import _cloneDeep from "lodash/cloneDeep";
+import _isArray from "lodash/isArray";
+import { Field } from "./Field";
+
+export class CustomField extends Field {
+  constructor({
+    fieldpath,
+    deserializedDefault = null,
+    serializedDefault = null,
+    allowEmpty = false,
+    vocabularyFields = [],
+  }) {
+    super({ fieldpath, deserializedDefault, serializedDefault, allowEmpty });
+    this.vocabularyFields = vocabularyFields;
+  }
+
+  _mapCustomFields(record, customFields, mapValue) {
+    if (customFields !== null) {
+      for (const [key, value] of Object.entries(customFields)) {
+        if (this.vocabularyFields.includes(key)) {
```

Comment:
Does it mean it is a vocabulary field? If yes:
```suggestion
        const isVocabularyField = this.vocabularyFields.includes(key);
        if (isVocabularyField) {
```